### ISSUE TITLE
Hunter pet AI defensive mode issue

### ISSUE THREAD

**Issue by jimmytaker** [slowtorta/turtlewow-bug-tracker#2879]
**Please, provide a clear description what the bug is:**

The pet only seems to be able to have 1 mob on its aggro table.

**Steps to reproduce the behavior with as much detail as possible:**

1. Put pet in defensive mode.
2. Send pet into a pack of 2 or more mobs. Attacking mob 1. (Both mobs attack the pet)
3. Assist the pet to kill mob 1.
-> After mob 1 dies the pet starts running back to the hunter and does not immediately acquire mob 2 as its target.

REMARK: The pet will stop and attack mob 2, if mob 2 attacks it again while it is running back to the hunter -> hence my assumption that it only has 1 mob on its aggro table.

**Expected behavior. Describe how it should work:**

The pet should not move back to the hunter after mob 1 dies (this is the expected behavior only when the pet is on passive). On defensive it should instead immediately turn to attack mob 2 (the mob with the highest threat on the pets aggro table).

**Comment by redmagejoe** [comment]
Can confirm this behavior. The pet behaves very inorganically and basically ignores its Hunter being attacked until it finishes with mob 1 and runs all the way back to the Hunter's side, like an NPC resetting to out of combat. Only after arriving beside the Hunter will it acknowledge the "new" target.

**Comment by Zeyla-TW** [comment]
I believe this may have been working as intended during 1.12.1 but defensive pet stance was revamped in later expansions to be more akin to what you're describing. It is difficult to compare though since Classic uses a more modern version of pet behaviors since it was down-ported from a Legion client.

**Comment by jimmytaker** [comment]
I mained a hunter on Lightshope's Darrowshire pve server before Classic. I am still using my own macro library from back then which played a huge part of my soloing every lvling instance at the lvl of its last boss back then. Pet targeting in defensive mode was what I used to quickly change targets after one died, extensively making use of macros of the sort "if UnitExists("pettarget") then TargetUnit("pettarget") end" . The way the pet acts on Turtle atm, these are not anywhere close as useful.

So while I do not know about hunter pet AI on 1.12, Lightshope's Darrowshire and Classic defensive pet AI had aggro table more than 1 mob long.
